[DECLUTTER WHY] Rework Privacy in Bitcoin (#569)
zkSNACKs/WasabiDoc: docs/why-wasabi/Coins.md

---
{
  "title": "Coins",
  "description": "On the nuances of Bitcoin's coin model with unspent transaction outputs, the privacy problems and how to fix it. This is the Wasabi documentation, an archive of knowledge about the open-source, non-custodial and privacy-focused Bitcoin wallet for desktop."
}
---

# Coins

Bitcoin has an accounting model of [unspent transaction output [UTXO]](https://bitcoin.org/en/blockchain-guide#introduction).

> A transaction has inputs: the coins that are spent, and outputs: the coins that are received.

Reviewer: ```suggestion
A transaction has inputs: the coins that are being spent, and outputs: the addresses that the coins that are being spent to.
```

Author: ```suggestion
A transaction has inputs: the coins which are being spent, and outputs: the addresses where the coins are being spent to.
```

Reviewer 2: How about this:
`A transaction has inputs: the coins that are being spent, and outputs: the corresponding newly created coins (unspent)`

Author: ```suggestion
A transaction has inputs: the coins that are being spent, and outputs: the corresponding newly created coins (unspent).
```

> A transaction has inputs: the coins that are spent, and outputs: the coins that are received.

Reviewer: ```suggestion
A transaction has inputs: the coins which are being spent, and outputs: the addresses to where the coins are being received.
```
I think this is more correct, grammatically, but yeah....sounds good to me.

The input of a transaction has to be an unspent output of a previous transaction.
Each UTXO is the tip of the chain of links between inputs and outputs, all the way back to a [coinbase transaction](https://en.bitcoin.it/wiki/Coinbase) that pays the miner.

## Problem

#### UTXOs are not fungible

Each UTXO is a unique snowflake with a public transaction history.
For example, when Alice sends a coin to Bob, then Bob does not just have any random UTXO, but he has specifically the coin that Alice has sent him.

> When Bob sends this coin to Charlie, then Charlie can look back two hops to see the transaction from Alice to Bob.

Reviewer: ```suggestion
When Bob sends this coin to Charlie, then Charlie can check the history of the coin and see the transaction from Alice to Bob.
```
This is perhaps nit-picky, but my concern is that someone new to bitcoin that might read this may not understand the practical meaning of "look back two hops".

Reviewer 2: Also we should make it clear that Charlie doesn't know the transaction was between Alice and Bob (unless he knows Alice's address).

> When Bob sends this coin to Charlie, then Charlie can look back two hops to see the transaction from Alice to Bob.

Reviewer: ```suggestion
When Bob sends this coin to Charlie, then Charlie can check the history of the coin and see every transaction that the coin has been a part of.
If Charlie knows Alice's address, then he will know that the coins that Bob sent him came from Alice.
```
Is this better?

Further, when Alice has one non-private coin and one private coin, and she selects both of them as the inputs of a transaction, the linking of these two coins strongly suggests that the coin that was private also belongs to Alice.
This means that coin consolidation can lead to an overall decrease of privacy, especially when using an automatic coin selection algorithm.

## Wasabi's Solution

#### Manual coin labeling and selection

> Contrary to many other wallets, Wasabi does not show only the total value of bitcoins in the wallet.

Reviewer 3: ```suggestion
Contrary to many other wallets, Wasabi does not show only the total value of bitcoin in the wallet.
```
This is the only occurrence of `bitcoins`, in all the other few cases it's spelled `bitcoin`.

Each UTXO is the tip of the chain of links between inputs and outputs, all the way back to a [coinbase transaction](https://en.bitcoin.it/wiki/Coinbase) that pays the miner.

## Problem

_**UTXOs are not fungible**_

Each UTXO is a unique snowflake with a public transaction history.
For example, when Alice sends a coin to Bob, then Bob does not just have any random UTXO, but he has specifically the coin that Alice has sent him.
When Bob sends this coin to Charlie, than Charlie can look back two hops to see the transaction from Alice to Bob.
If Bob does not want Charlie to know that the transaction between Alice and Bob happened, then Bob needs to send Charlie a different coin.

Further, when Alice has one non-private coin and one private coin, and she selects both of them as the input of one transaction, the linking of these two coins strongly suggests that the coin that was private also belongs to Alice.
This means that coin consolidation can lead to an overall decrease of privacy, especially when using an automatic coin selection algorithm.

## Wasabi's Solution

_**Manual coin labeling and selection**_

> Contrary to many other wallets, Wasabi does not show only the total value of bitcoins in the wallet. 

Reviewer 2: ```suggestion
Contrary to many other wallets, Wasabi does not show only the total value of bitcoins in the wallet.
```

Rather, in both the `Send` and `CoinJoin` tabs there is a list of all the individual UTXOs.
Because Wasabi requires users to label every receiving address, the history of each coin is clear at first glance.

> In order to spend a specific coin, it must be manually selected, which prevents the wrong coin from being included in the transaction.

Reviewer 2: ```suggestion
In order to spend a specific coin, it must be manually selected, which prevents the wrong coin from being included in the transaction.

```

Each UTXO is a unique snowflake with a public transaction history.
For example, when Alice sends a coin to Bob, then Bob does not just have any random UTXO, but he has specifically the coin that Alice has sent him.

> When Bob sends this coin to Charlie, then Charlie can check the history of the coin and see the transaction from Alice to Bob.

Reviewer 2: Charlie knows that the transaction is to Bob but he doesn't know that the transaction is from Alice (unless he knows Alice's address).
